src/CplexSolverInterface.jl

```diff
@@ -449,22 +449,32 @@ function mastercallback(env::Ptr{Void}, cbdata::Ptr{Void}, wherefrom::Cint, user
     cpxrawcb = CallbackData(cbdata, model.inner)
     if wherefrom == CPX_CALLBACK_MIP_CUT_FEAS || wherefrom == CPX_CALLBACK_MIP_CUT_UNBD
         state = :MIPSol
-        cpxcb = CplexLazyCallbackData(cpxrawcb, state, wherefrom, userinteraction_p)
-        if model.lazycb != nothing
-            stat = model.lazycb(cpxcb)
-            if stat == :Exit
-                terminate(model.inner)
-            end
-        end
     # elseif wherefrom == CPX_CALLBACK_MIP_CUT_LOOP || wherefrom == CPX_CALLBACK_MIP_CUT_LAST
     elseif wherefrom == CPX_CALLBACK_MIP_CUT_LAST
         state = :MIPNode
+    else
+        state = :Intermediate
+    end
+
+    if model.infocb != nothing
+        cpxcb = CplexInfoCallbackData(cpxrawcb, state, wherefrom)
+        stat = model.infocb(cpxcb)
+        if stat == :Exit
+            terminate(model.inner)
+        end
+    end
+    if model.lazycb != nothing && state == :MIPSol
+        cpxcb = CplexLazyCallbackData(cpxrawcb, state, wherefrom, userinteraction_p)
```

Comment:
Oh, nevermind, I missed line 461 where the callback function was actually called.